WordPress/Classes/ApplicationToken/ApplicationTokenListView.swift

```diff
@@ -0,0 +1,149 @@
+import SwiftUI
+import DesignSystem
+import WordPressUI
+import WordPressAPI
+
+struct ApplicationTokenListView: View {
+
+    @ObservedObject
+    private var viewModel: ApplicationTokenListViewModel
+
+    fileprivate init(tokens: [ApplicationTokenItem]) {
+        let dataProvider = StaticTokenProvider(tokens: .success(tokens))
+        self.init(viewModel: ApplicationTokenListViewModel(dataProvider: dataProvider))
+    }
+
+    fileprivate init(error: Error) {
+        let dataProvider = StaticTokenProvider(tokens: .failure(error))
+        self.init(viewModel: ApplicationTokenListViewModel(dataProvider: dataProvider))
+    }
+
+    init(viewModel: ApplicationTokenListViewModel) {
+        self.viewModel = viewModel
+    }
+
+    var body: some View {
+        VStack {
+            if viewModel.isLoadingData {
+                ProgressView()
+            } else if let error = viewModel.errorMessage {
+                EmptyStateView(Self.errorTitle, systemImage: "exclamationmark.triangle", description: error)
+            } else {
+                List(viewModel.applicationTokens) { token in
+                    ApplicationTokenListItemView(item: token)
+                }
+                .listStyle(.plain)
+            }
+        }
+        .navigationTitle(Self.title)
+        .navigationBarTitleDisplayMode(.inline)
+        .onAppear {
```

Comment:
(nit) I would suggest using the [task()](https://developer.apple.com/documentation/swiftui/view/task(priority:_:)) modifier and checking `!applicationTokens.isEmpty` in the ViewModel.

Reply:
I tried that. But the `task` closure would get called upon every view updates. I'm going to give it another try, just to make sure...